nixos/default.nix

```diff
@@ -7,19 +7,54 @@
 }:
 
 let
-
   cfg = config.home-manager;
 
-  serviceEnvironment =
-    lib.optionalAttrs (cfg.backupFileExtension != null) {
-      HOME_MANAGER_BACKUP_EXT = cfg.backupFileExtension;
-    }
-    // lib.optionalAttrs cfg.verbose { VERBOSE = "1"; };
+  baseService = username: {
+    Type = "oneshot";
+    RemainAfterExit = "yes";
+    TimeoutStartSec = "5m";
+    SyslogIdentifier = "hm-activate-${username}";
+  };
+
+  baseUnit = username: {
+    description = "Home Manager environment for ${username}";
+    stopIfChanged = false;
+    serviceConfig = baseService username;
+    environment =
+      {
+        # needed to run qt programs like kwriteconfig
+        QT_QPA_PLATFORM = "offscreen";
+      }
+      // lib.optionalAttrs (cfg.backupFileExtension != null) {
+        HOME_MANAGER_BACKUP_EXT = cfg.backupFileExtension;
+      }
+      // lib.optionalAttrs cfg.verbose {
+        VERBOSE = "1";
+      };
+  };
 
+  # we use a service separated from nixos-activation
+  # to keep the logs separate
+  hmDropIn = "/share/systemd/user/home-manager.service.d";
 in
 {
   imports = [ ./common.nix ];
 
+  options.home-manager.useUserService = lib.mkEnableOption ''
+    Activate each user's environment only when they log in, using a
```

Comment:
I beleieve the generated doc prepends "Enable" so that wont render great. maybe use `mkOption` instead.